### PR TITLE
[EA Forum only] Debate week home page banner

### PR DESCRIPTION
**Pull request by s-cheng**
This PR sets up a `ForumEvents` variation that includes a user poll. [Figma here.](https://www.figma.com/design/qAHRGADVOGojqKpqbGMFFu/2024-Q2?node-id=1235-3130&m=dev)

There are a lot of differences between this event and the existing `ForumEvents` (the UIs are more different than similar), but I think it's worth combining them because it allows admins to manage events using the same tool - from the admin perspective they are similar. And I expect we will want to do more debate/poll-style events in the future. However, since the AI Welfare Debate Week is starting on July 1, and since this is the first event of its type (so I expect it to change in the future), I didn't fully build out the tool functionality.

This is what a logged out user sees when they arrive on the site. They can drag the vote icon but when they release, it will just reset and open the login modal.
Screen Shot 2024-06-27 at 12 06 20 AM

Here's a screenshot of the banner expanded, and a logged in user hovering over the vote icon:
Screen Shot 2024-06-27 at 12 10 04 AM

~After dragging and dropping your initial vote (or after clicking "view results"), we show you everyone's votes.~ Chatted with Toby and we changed this so that you only see other votes if you click on "view results". This is so people who want to vote without being influenced can do so. You can click the little "x" to clear out your vote.
Screen Shot 2024-06-27 at 12 10 44 AM

If you change your vote, we'll show you this modal and ask if you want to give points to any posts that changed your mind:
Screen Shot 2024-06-27 at 3 07 40 PM

On mobile, we just show a banner:
Screen Shot 2024-06-26 at 3 33 55 PM

-----
TODO approx in priority order:
1. Post page UI for debate week (separate PR branched off of this one)
2. Adjust the voting slider to work for screens <1000px (I will prob do this in the post page PR)
3. Give users a way to see voters when their votes are bunched together (doesn't seem like a blocker for deploying this)

Known issues that I prob won't bother to fix:
1. Things are slightly misaligned in Safari
2. Post authors can give their own post points (the points don't really matter and Toby is fine with us ignoring this, plus I think this could be a legit case if, for example, one of the coauthors updates based on parts of the post that they didn't write)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207673009856845) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx [hunk 1]
```diff
@@ -69,10 +157,41 @@ const styles = (theme: ThemeType) => ({
...
   image: {
     position: "absolute",
     zIndex: -1,
-    top: "-45%",
+    top: -HEADER_HEIGHT,
+    right: 0,
+    width: '100%',
+  },
+  imageWithGradient: {
+    // top: "-45%",
```
Comment (s-cheng) [inline packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx:194]: Oops I wanted to test this, will do so tomorrow

File: packages/lesswrong/lib/collections/forumEvents/schema.ts [hunk 2]
```diff
@@ -59,6 +67,38 @@ const schema: SchemaType<"ForumEvents"> = {
...
+   */
+  publicData: {
+    type: Object,
+    blackbox: true,
+    optional: true,
+    nullable: true,
+    hidden: true,
+    canRead: ["guests"],
+    canCreate: ["members"],
+  },
```
Comment (s-cheng) [inline packages/lesswrong/lib/collections/forumEvents/schema.ts:95]: Not super happy about this one, but I figured we might want to accommodate events similar to the good news banner, where there is a different sort of publicly visible event data 🤷‍♀️

**Review by oetherington — APPROVED** [review]
👍

File: packages/lesswrong/components/common/Header.tsx [hunk 3]
```diff
@@ -416,6 +422,28 @@ const Header = ({
       />
     );
 
+  let headerStyle: CSSProperties = {}
+  const bannerImageId = currentForumEvent?.bannerImageId
+  // On EAF, forum events with polls also update the home page header background
+  if (currentRoute?.name === 'home' && bannerImageId && currentForumEvent.includesPoll && isEAForum) {
```
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/components/common/Header.tsx]: Thanks, didn't remember we had that - updated!

File: packages/lesswrong/lib/collections/forumEvents/schema.ts [hunk 4]
```diff
@@ -59,6 +67,38 @@ const schema: SchemaType<"ForumEvents"> = {
...
+  }}
+   */
+  publicData: {
+    type: Object,
+    blackbox: true,
+    optional: true,
+    nullable: true,
+    hidden: true,
+    canRead: ["guests"],
+    canCreate: ["members"],
```
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/lib/collections/forumEvents/schema.ts:94]: Yeah I was thinking that it's not necessary to have that, so we should default to stricter permissions (i.e. we wouldn't want users to arbitrarily update the value of this field)

File: packages/lesswrong/components/common/Header.tsx [hunk 5]
```diff
@@ -416,6 +422,28 @@ const Header = ({
...
+    const darkColor = currentForumEvent?.darkColor
+    const background = `top / cover no-repeat url(${makeCloudinaryImageUrl(bannerImageId, {
+      c: "fill",
+      dpr: "auto",
+      q: "auto",
+      f: "auto",
+      g: "north",
+    })})${darkColor ? `, ${darkColor}` : ''}`
+    headerStyle = {background}
```
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/components/common/Header.tsx:438]: No reason, I just did it quickly - I'll update it

File: packages/lesswrong/components/common/ForumIcon.tsx [hunk 6]
```diff
@@ -218,6 +225,7 @@ export type ForumIconName =
   "ClipboardDocumentList" |
   "ClipboardDocument" |
   "QuestionMarkCircle" |
+  "QuestionMarkCircleOutline" |
```
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/components/common/ForumIcon.tsx]: Thanks for flagging! I merged in master and used your version 👍

File: packages/lesswrong/lib/collections/forumEvents/schema.ts [hunk 7]
```diff
@@ -59,6 +67,38 @@ const schema: SchemaType<"ForumEvents"> = {
     type: String,
     control: "ImageUpload",
   },
+  includesPoll: {
+    ...defaultProps(),
+    optional: true,
+    nullable: true,
```
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/lib/collections/forumEvents/schema.ts]: Oh right I meant to do that and forgot, thanks for the reminder!

File: packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx [hunk 8]
```diff
@@ -109,38 +236,175 @@ export const ForumEventFrontpageBanner = ({classes}: {
...
+              <div className={classes.titleWithPoll}>{title}</div>
+              <div className={classes.dateWithPoll}>{date}</div>
+              <div className={classes.descriptionWithPoll}>{description}.</div>
+            </div>
+          </div>
+        </div>}
+        <div className={classes.contentWithPollMobile}>
+          <div className={classes.titleWithPollMobile}>{title}</div>
+          <div className={classes.dateWithPoll}>{date}</div>
+          <div className={classes.descriptionWithPoll}>{description}, and vote on desktop.</div>
```
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx]: Yup I was being lazy, but since I have to re-do the migration to merge master in anyway, I went ahead and added a `frontpageDescriptionMobile` field

File: packages/lesswrong/server/resolvers/forumEventResolvers.ts [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+
+addGraphQLResolvers({
+  Mutation: {
+    AddForumEventVote: async (
+      _root: void,
+      {forumEventId, x, delta, postIds}: {forumEventId: string, x: number, delta?: number, postIds?: string[]},
+      {currentUser, repos}: ResolverContext
+    ) => {
+      if (!currentUser) {
```
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/server/resolvers/forumEventResolvers.ts:12]: Hmm yeah I'll ask Toby, but he didn't seem bothered by voting edge cases (like authors voting for their own posts). I think he views this event as something of a test run for the format, and the actual votes here are not really consequential. I'll leave this as-is unless he requests a change.

File: packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx (hunk 1 above)
Comment (s-cheng) [reply packages/lesswrong/components/forumEvents/ForumEventFrontpageBanner.tsx]: Ok looking at a standard forum event with and without this, I think Agnes just added it to adjust a specific banner, so I'm going to remove it since it's not a general style we want
